### PR TITLE
Integration with report server

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java [hunk 1]
```diff
@@ -42,14 +43,16 @@
...
+                                                   @ApiParam(value = "reportName") @RequestParam(name = "reportName", required = false) String reportName,
                                                    @RequestBody(required = false) String loadflowParams) {
         LoadFlowParameters parameters = loadflowParams != null
                 ? JsonLoadFlowParameters.read(new ByteArrayInputStream(loadflowParams.getBytes()))
                 : null;
 
         List<UUID> otherNetworksUuid = otherNetworks != null ? otherNetworks.stream().map(UUID::fromString).collect(Collectors.toList()) : Collections.emptyList();
 
-        LoadFlowResult result = loadFlowService.loadFlow(networkUuid, otherNetworksUuid, parameters, provider);
+        LoadFlowResult result = loadFlowService.loadFlow(networkUuid, otherNetworksUuid, parameters, provider, Optional.ofNullable(reportId), Optional.ofNullable(reportName));
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java:55]: Usually, optional is used to return a value not as an argument

Comment (geofjamg) [reply src/main/java/org/gridsuite/loadflow/server/LoadFlowController.java:55]: Optional as an argument is more that unusual, it is FORBIDDEN !

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java [hunk 2]
```diff
@@ -45,37 +74,56 @@ private Network getNetwork(UUID networkUuid) {
...
+            if (doReport) {
+                sendReport(reportId.get(), reporter, false);
+            }
         }
-
         return result;
     }
+
+    private void sendReport(UUID networkUuid, Reporter reporter, boolean overwrite) {
+        var restTemplate = new RestTemplate();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java:118]: Why not use a web flux client

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java [hunk 3]
```diff
@@ -45,37 +74,56 @@ private Network getNetwork(UUID networkUuid) {
...
     LoadFlowResult loadFlow(UUID networkUuid, List<UUID> otherNetworksUuid, LoadFlowParameters parameters,
-                            String provider) {
+                            String provider, Optional<UUID> reportId, Optional<String> reportName) {
         LoadFlowParameters params = parameters != null ? parameters : new LoadFlowParameters();
         LoadFlowResult result;
 
         LoadFlow.Runner runner = LoadFlow.find(provider != null ? provider : DEFAULT_PROVIDER);
+        Reporter reporter;
         if (otherNetworksUuid.isEmpty()) {
             Network network = getNetwork(networkUuid);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java:84]: Why it is not  an option the report like merging view ?

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java [hunk 4]
```diff
@@ -45,37 +74,56 @@ private Network getNetwork(UUID networkUuid) {
...
+        var restTemplate = new RestTemplate();
+        var headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        var resourceUrl = reportServerURI + DELIMITER + REPORT_API_VERSION + DELIMITER + "report" + DELIMITER + networkUuid.toString();
+        var uriBuilder = UriComponentsBuilder.fromHttpUrl(resourceUrl).queryParam("overwrite", overwrite);
+        try {
+            restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.PUT, new HttpEntity<>(objectMapper.writeValueAsString(reporter), headers), ReporterModel.class);
+        } catch (Exception error) {
+            LOGGER.error(error.getMessage());
+        }
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java:127]: throw new PowsyblException(error); ?
To have a feedback to the front

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java:94]: done

File: src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java:135]: done

**Review by geofjamg — CHANGES_REQUESTED** [review]
what i the purpose of overwriteReport, it does not seem to be used?

**Review by geofjamg — APPROVED** [review]

**Pull request merged**